test/jdk/java/net/httpclient/PostFromGetTest.java

```diff
@@ -0,0 +1,605 @@
+    }
+
+    private <T> void testGetPost(String uri, boolean sameClient,
+                            Supplier<BodyHandler<T>> handleSupplier,
+                            Function<T, BodyPublisher> publisherSupplier)
+            throws Exception {
+        checkSkip();
+        HttpClient client = null;
+        int size = 64 * 1024 * 64;
+        String geturi = uri + "/get/?" + size;
+        String posturi = uri + "/post/?" + size;
+        out.printf("%n%n%s testGetSendAsync(%s, %b)%n", now(), uri, sameClient);
+        for (int i=0; i< ITERATION_COUNT; i++) {
+            if (!sameClient || client == null)
+                client = newHttpClient(sameClient);
+            Tracker tracker = TRACKER.getTracker(client);
+
+            HttpRequest getreq = HttpRequest.newBuilder(URI.create(geturi))
+                    .GET()
+                    .build();
+            HttpResponse<T> getresponse =
+                    client.send(getreq, handleSupplier.get());
+
+            HttpRequest postreq = HttpRequest.newBuilder(URI.create(posturi))
+                    .POST(publisherSupplier.apply(getresponse.body()))
+                    .build();
+            HttpResponse<InputStream> postresponse = client.send(postreq, BodyHandlers.ofInputStream());
+            int count = 0;
+            try (var is = postresponse.body()) {
+                int c;
+                while ((c = is.read()) >= 0) {
+                    assertEquals(c, count % 256);
+                    count++;
+                    if (count % (8*1024) == 0) {
+                        out.printf("Post response: %s/%s bytes, %s%%%n",
+                                count, size, (long)((((double)count)/((double)size))*100f));
+                    }
+                }
+                assertEquals(count, size);
+            }
+
+            var error = TRACKER.check(tracker, 1000,
+                    (t) -> t.getOutstandingOperations() > 0 || t.getOutstandingSubscribers() > 0,
+                    "subscribers for testGetSendAsync(%s)\n\t step [%s]".formatted(uri, i),
+                    false);
+            Reference.reachabilityFence(client);
+            if (error != null) throw error;
+        }
+        assert client != null;
+        if (!sameClient) client.close();
+    }
+
+
+
+    @BeforeTest
+    public void setup() throws Exception {
+        sslContext = new SimpleSSLContext().get();
+        serverExecutor = Executors.newCachedThreadPool
+                (Thread.ofPlatform().daemon().name("Server").factory());
+        if (sslContext == null)
+            throw new AssertionError("Unexpected null sslContext");
+
+        // HTTP/1.1
+        HttpTestHandler h1_chunkHandler = new H1GetPostHandler();
+        httpTestServer = HttpTestServer.create(HTTP_1_1, null, serverExecutor);
+        httpTestServer.addHandler(h1_chunkHandler, "/http1/x/");
+        httpURI = "http://" + httpTestServer.serverAuthority() + "/http1/x/";
```

Comment:
There has been occasions where a request got incorrectly served by a parallel running test. Last time we've salted the paths:

https://github.com/openjdk/jdk/blob/f529bf712d8946584999dfc98abea60c22c97167/test/jdk/java/net/httpclient/PlainProxyConnectionTest.java#L77

We can consider doing the same here too.

Reply:
replaced `x` with `PostFromGet`. That should be enough.